include/dlaf/util_matrix.h

```diff
@@ -40,5 +40,31 @@ void assert_blocksize_square(const Matrix& matrix, std::string function, std::st
                                 " is not square.");
 }
 
+/// @brief Verify if dlaf::Matrix is on local memory
+///
+/// @tparam Matrix refers to a dlaf::Matrix object
+/// @throws std::invalid_argument if the matrix is not stored on local memory
+template <class Matrix>
+void assert_local_matrix(const Matrix& matrix, std::string function, std::string mat_name) {
+  assert(matrix.size().isValid());
```

Comment:
I'm not so sure that this assert has to be here.

Why do you need this check in debug?

Reply:
Side note: The size of a matrix is always valid. The only way to have an invalid size is to manipulate the memory in which the matrix members are stored.